homeassistant/components/ambient_station/config_flow.py

```diff
@@ -58,9 +58,19 @@ class AmbientStationFlowHandler(config_entries.ConfigFlow):
             user_input[CONF_API_KEY], user_input[CONF_APP_KEY], session)
 
         try:
-            await client.api.get_devices()
+            devices = await client.api.get_devices()
         except AmbientError:
             return await self._show_form({'base': 'invalid_key'})
 
+        if not devices:
+            return await self._show_form({'base': 'no_devices'})
+
+        # Although the application key identifies the config entry, it's way
+        # too long to show nicely in the UI. Even though an account may have
```

Comment:
Would it be an option to show the first X characters of the application key as title, similar to how a commit sha is shortened in the github UI?

Reply:
You bet – that's definitely more representative than the name of the first device without being obnoxiously long.